harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -374,7 +374,7 @@ def _auto_step_lr_scheduler_per_batch(
     def _should_update_scaler(self) -> bool:
         if not self.context._scaler or not self.context.experimental._auto_amp:
             return False
-        if self.context.distributed.size > 1:
+        if self.context.distributed.size > 1 or self.context._aggregation_frequency > 1:
             return self.context._should_communicate_and_update()  # type: ignore
         return True
```

Comment:
Why not just return `self.context._should_communicate_and_update()` like you had it before?  It was simpler and it should be equivalent.

Semantically, this is exactly what `_should_communicate_and_update()` means.